src/actix/helpers.rs

```diff
@@ -141,6 +150,38 @@ fn log_service_error(err: &StorageError) {
 #[error("{0}")]
 pub struct HttpError(StorageError);
 
+impl HttpError {
+    fn headers(&self) -> HeaderMap {
+        let mut headers = HeaderMap::new();
+        match &self.0 {
+            StorageError::RateLimitExceeded {
+                description: _,
+                retry_after,
+            } => {
+                if let Some(retry_after) = retry_after {
+                    // Retry-After is expressed in seconds `https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After`
+                    headers.insert(
+                        header::RETRY_AFTER,
+                        header::HeaderValue::from(retry_after.as_secs()),
```

Comment:
`as_secs()` rounds down. I don't think that is desired, and we should ceil instead.

Maybe:

```suggestion
                        header::HeaderValue::from(retry_after.as_secs_f32().ceil() as u32),
```

Or something like:

```suggestion
                        header::HeaderValue::from(retry_after.as_secs() + (retry_after.subsec_nanos() > 0) as u64),
```

I find [MDNs page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After#directives) quite unclear on whether the header supports subsecond values. Based on [this](https://httpwg.org/specs/rfc9110.html#field.retry-after) I don't think it does, and we should ceil.

Reply:
Implemented the first suggestion.